### PR TITLE
Allow running resstock in Parallel

### PR DESCRIPTION
**Pull request by jmarrec**
## Pull Request Description

Trying to upstream at https://github.com/NREL/resstock/pull/1131

## Checklist

Not all may apply:

- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] Changelog has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: measures/BuildExistingModel/measure.rb [hunk 1]
```diff
@@ -243,7 +249,11 @@ def run(model, runner, user_arguments)
     measures_dir = File.join(File.dirname(__FILE__), '../../measures')
     hpxml_measures_dir = File.join(File.dirname(__FILE__), '../../resources/hpxml-measures')
     lookup_file = File.join(resources_dir, 'options_lookup.tsv')
-    buildstock_csv_path = File.absolute_path(File.join(characteristics_dir, 'buildstock.csv')) # Should have been generated by the Worker Initialization Script (run_sampling.rb) or provided by the project
+
+    buildstock_csv_path = args['buildstock_csv_path']
+    unless (Pathname.new buildstock_csv_path).absolute?
+      buildstock_csv_path = File.absolute_path(File.join(characteristics_dir, buildstock_csv_path)) # Should have been generated by the Worker Initialization Script (run_sampling.rb) or provided by the project
```
Comment (jason-curtis) [inline measures/BuildExistingModel/measure.rb:255]: The buildstock_csv arg can already be specified by the resstock project file, right? So technically you could just use a bunch of different project files, one for each buildstock?
This does seem more convenient but I'm wondering if there's a different reason here that I'm missing.

**Review by jason-curtis — APPROVED** [review]
I see most of the conversation is happening on the upstream PR - good stuff there, I'm glad you're getting a response!

File: measures/BuildExistingModel/measure.rb (hunk 1 above)
Comment (jmarrec) [reply measures/BuildExistingModel/measure.rb:255]: Yes. For us that means no more copy from run folder to the resstock/lib


The yaml has a sample file param. But run_analysis.rb just copied it. BuildExistingModel measure plain assumed it was in the resstock/lib folder

**Comment by jason-curtis** [comment]
Got it. The change seems very important then!

File: workflow/run_analysis.rb [hunk 2]
```diff
@@ -439,11 +448,50 @@ def run_workflow(yml, n_threads, measures_only, debug_arg, overwrite, building_i
...
+    [
+      rel_path,
+      File.mtime(File.join(directory_path, rel_path)), # mtime is affected by the copy, but we passed preserve = true
+      File.size(File.join(directory_path, rel_path))
+    ]
+  }.to_s)
+  return dir_checksum
+end
+
+def create_lib_folder(lib_dir, resources_dir, housing_characteristics_dir, debug: false)
```
Comment (jason-curtis) [inline workflow/run_analysis.rb:463]: It looks like all we're doing here is mirroring `resources_dir` and `housing_characteristics_dir` into this lib directory. Why do we have to do that?

Comment (jmarrec) [reply workflow/run_analysis.rb:463]: :man_shrugging: . Maybe this was "easier" for resstock so all measures didn't have to carry two extra paths and could just use whatever is inside lib/. I don't know.

**Pull request merged**